developer/debugging/test-bench.md

```diff
@@ -15,57 +15,71 @@ This guide shows how to set up simple test bench that automatically test your co
+### Enabling Network Access in Dom0
 
-# adjust this for your NIC (run lspci)
-BDF=0000:02:00.0
-
-prog=$(basename $0)
-
-pciunbind() {
-    local path
-    path=/sys/bus/pci/devices/${1}/driver/unbind
-    if ! [ -w ${path} ]; then
-        echo "${prog}: Device ${1} not bound"
-        return 1
-    fi
-    echo -n ${1} >${path}
-}
-
-pcibind() {
-    local path
-    path=/sys/bus/pci/drivers/${2}/bind
-    if ! [ -w ${path} ]; then
-        echo "${prog}: Driver ${2} not found"
-        return 1
-    fi
-    echo ${1} >${path}
-}
-
-pciunbind ${BDF}
-pcibind ${BDF} e1000e
-
-dhclient
-```
+Internet access is intentionally disabled by default in dom0. But to ease the deployment process we will give it access. The following steps should be done in `dom0`.
 
-TODO: describe how to run this at every startup
+> **Note:** the following assume you have only one network card. If you have two, pick one and leave the other attached to `sys-net`.
 
-Now configure your DHCP server so your testbench gets static IP and connect your machine to your local network. You should ensure that your testbench can reach the Internet.
+1. Remove the network card (PCI device) from `sys-net`
+2. Restart your computer (for the removal to take effect)
+3. The following script should enable your network card in dom0. *Be sure to adjust the script's variables to suit your needs.* You'll need to run this at every startup (TODO: describe how to run this at every startup).
 
-Install `openssh-server` on your testbench:
+    ```bash
+    #!/bin/sh
 
-~~~
-yum install openssh-server
-~~~
+    # adjust this for your NIC (run lspci)
+    BDF=0000:02:00.0
+
+    # adjust this for your network driver
+    DRIVER=e1000e
 
-Ensure that sudo works without password from your user account (it should by default).
+    prog=$(basename $0)
+    
+    pciunbind() {
+        local path
+        path=/sys/bus/pci/devices/${1}/driver/unbind
+        if ! [ -w ${path} ]; then
+            echo "${prog}: Device ${1} not bound"
+            return 1
+        fi
+        echo -n ${1} >${path}
+    }
+
+    pcibind() {
+        local path
+        path=/sys/bus/pci/drivers/${2}/bind
+        if ! [ -w ${path} ]; then
+            echo "${prog}: Driver ${2} not found"
+            return 1
+        fi
+        echo ${1} >${path}
+    }
+
+    pciunbind ${BDF}
+    pcibind ${BDF} ${DRIVER}
+    
+    sleep 1
+    dhclient
+    ```
+
+4. Configure your DHCP server so your testbench gets static IP and connect your machine to your local network. You should ensure that your testbench can reach the Internet.
+
+5. Install `openssh-server` on your testbench.
+ 
+    ~~~
+    sudo dnf --setopt=reposdir=/etc/yum.repos.d install openssh-server
+    ~~~
+
+> **Note:** If you want to install additional software in dom0 and your only network card was assigned to dom0, then _instead_ of the usual `sudo qubes-dom0-update install <PACKAGE>` now you run `sudo dnf --setopt=reposdir=/etc/yum.repos.d install <PACKAGE>`.
```

Comment:
Almost. `qubes-dom0-update` has no `install` argument.

Reply:
Ah. You're right! Sorry about that.